adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m

```diff
@@ -0,0 +1,248 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMediationVungleNativeAd.h"
+#include <stdatomic.h>
+#import "GADMAdapterVungleUtils.h"
+#import "GADMAdapterVungleRouter.h"
+#import <VungleSDK/VungleNativeAd.h>
+
+@interface GADMediationVungleNativeAd () <GADMediationNativeAd, VungleNativeAdDelegate, GADMAdapterVungleDelegate>
```

Comment:
Isn't that for native we only support waterfall mediation (not bidding)?

Reply:
Yes. Both flows will utilize the same class. There is no difference between the two.
The native ad implementation on Vungle's end utilizes an instance-based approach so it doesn't need the router to track its reference to relay callbacks to the ad object.